com.ibm.streamsx.json/impl/java/src/com/ibm/streamsx/json/converters/JSONToTupleConverter.java

```diff
@@ -0,0 +1,404 @@
+package com.ibm.streamsx.json.converters;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.ibm.json.java.JSONArray;
+import com.ibm.json.java.JSONObject;
+import com.ibm.streams.operator.Attribute;
+import com.ibm.streams.operator.StreamSchema;
+import com.ibm.streams.operator.Tuple;
+import com.ibm.streams.operator.Type;
+import com.ibm.streams.operator.logging.TraceLevel;
+import com.ibm.streams.operator.meta.CollectionType;
+import com.ibm.streams.operator.meta.TupleType;
+import com.ibm.streams.operator.types.RString;
+import com.ibm.streams.operator.types.Timestamp;
+
+public class JSONToTupleConverter {
+
+	private static Logger l = Logger.getLogger(JSONToTupleConverter.class.getCanonicalName());
+	
+	/**
+	 * Convert a JSON object to a Java object of the specified type. 
+	 * 
+	 * @param name The name of the attribute being converted (used for logging purposes)
+	 * @param type SPL type that the JSON object is being converted to. 
+	 * @param jsonObj JSON object that is being converted (can be either JSONObject or JSONArray) 
+	 * @param parentType This parameter is used when converting JSON arrays or maps to an attribute. For all other types, can be set to null.
+	 * @return Java object of the specified type representing the converted JSON object.
+	 * @throws Exception If there was a problem converting the JSON object.
+	 */
+	public static Object jsonToAttribute (String name, Type type, Object jsonObj, Type parentType) throws Exception {
+
+		if(l.isLoggable(TraceLevel.DEBUG)) {
+			l.log(TraceLevel.DEBUG, "Converting: " + name + " -- " + type.toString());
+		}
+		if(jsonObj == null) return null;
+		try {
+			switch(type.getMetaType()) {
+			case INT8:
+			case UINT8:
+				if(jsonObj instanceof Number)
+					return ((Number)jsonObj).byteValue();
+				return Byte.parseByte(jsonObj.toString());
+			case INT16:
+			case UINT16:
+				if(jsonObj instanceof Number)
+					return ((Number)jsonObj).shortValue();
+				return Short.parseShort(jsonObj.toString());
+			case INT32:
+			case UINT32:
+				if(jsonObj instanceof Number)
+					return ((Number)jsonObj).intValue();
+				return Integer.parseInt(jsonObj.toString());
```

Comment:
Does this work for unsigned values?

If the value is greater than Integer.MAX_VALUE I don't think the code will return the correct value, either in the String or the Number case?